src/sv-pipeline/04_variant_resolution/scripts/clean_vcf_part5_update_records.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+import argparse
+from collections import Counter
+import gzip
+import pysam
+import sys
+import svtk.utils as svu
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('revise_vcf_lines', type=argparse.FileType('r'))
+    parser.add_argument('normal_revise_vcf')
+    parser.add_argument('famfile', type=argparse.FileType('r'))
+    parser.add_argument('sexchr_revise')
+    parser.add_argument('multi_geno_ids_txt')
+    parser.add_argument('outlier_samples_list', type=argparse.FileType('r'))
+    args = parser.parse_args()
+
+    # load the revised lines and index by ID
+    revised_lines_by_id = {}
+    with pysam.VariantFile(args.revise_vcf_lines) as revise_vcf:
+        header2 = revise_vcf.header
+        for record in revise_vcf:
+            revised_lines_by_id[record.id] = record
+    print("loaded {} revised lines".format(len(revised_lines_by_id)), file=sys.stderr)
+
+    outlier_samples = set([line.rstrip() for line in args.outlier_samples_list if not line.isspace()])
+    print("loaded {} outlier samples".format(len(outlier_samples)), file=sys.stderr)
+
+    male_samples = set()
+    for line in args.famfile:
+        if line.isspace():
+            continue
+        fields = line.rstrip().split("\t")
+        if fields[4] == '1':
+            male_samples.add(fields[1])
+    print("identified {} male samples".format(len(male_samples)), file=sys.stderr)
+
+    if args.sexchr_revise.endswith(".gz"):
+        sexchr_revise = set([line.rstrip() for line in gzip.open(args.sexchr_revise, 'rt')])
+    else:
+        sexchr_revise = set([line.rstrip() for line in open(args.sexchr_revise, 'rt')])
```

Comment:
```suggestion
        sexchr_revise = {line.rstrip() for line in gzip.open(args.sexchr_revise, 'rt')}
    else:
        sexchr_revise = {line.rstrip() for line in open(args.sexchr_revise, 'rt')}
```
Can avoid creating intermediate list